File: api/schemas/mcp.py
from pydantic import BaseModel

class MCPServerDetailRequest(BaseModel):
    id: int

class MCPServerCreateRequest(BaseModel):
    name: str
    category: int
    url: str | None = None
    cmd: str | None = None
    args: str | None = None
    env: str | None = None
    headers: str | None = None
    
class MCPServerUpdateRequest(BaseModel):
    id: int
    category: int | None = None
    name: str | None = None
    enable: bool | None = None
    url: str | None = None
    cmd: str | None = None
    args: str | None = None
    headers: str | None = None
    env: str | None = None
    
class MCPServerDeleteRequest(BaseModel):
    id: int
    
class MCPServerInfo(BaseModel):
    id: int
    name: str
    enable: bool
    category: int
    url: str | None = None
    cmd: str | None = None
    args: str | None = None
    env: str | None = None
    headers: str | None = None
    
class MCPServerSearchRequest(BaseModel):
    keyword: str | None = None
    
class MCPServerSearchResponse(BaseModel):
    data: list[MCPServerInfo]

File: web/src/service/mcp.ts
import mcpApi from '@/api/mcp'
import { MCPServerCreateRequest, MCPServerDeleteRequest, MCPServerDetailRequest, MCPServerInfo, MCPServerSearchRequest, MCPServerSearchResponse, MCPServerUpdateRequest, NormalResponse } from '@/generated'
import { request } from '@/lib/request'

export const createMCPServer = async (data: MCPServerCreateRequest): Promise<NormalResponse> => {
    return await request(mcpApi.createMCPServer, {
        data: data
    })
}

export const updateMCPServer = async (data: MCPServerUpdateRequest): Promise<NormalResponse> => {
    return await request(mcpApi.updateMCPServer, {
        data: data
    })
}

export const deleteMCPServer = async (data: MCPServerDeleteRequest): Promise<NormalResponse> => {
    return await request(mcpApi.deleteMCPServer, {
        data: data
    })
}

export const searchMCPServer = async (data: MCPServerSearchRequest): Promise<MCPServerSearchResponse> => {
    return await request(mcpApi.searchMCPServer, {
        data: data
    })
}

export const getMCPServerDetail = async (data: MCPServerDetailRequest): Promise<MCPServerInfo> => {
    return await request(mcpApi.getMCPServerDetail, {
        data
    })
}